uuid.gd

```diff
@@ -39,3 +43,94 @@ static func v4():
     # clock
     b[10], b[11], b[12], b[13], b[14], b[15]
   ]
+  
+static func v4_rng(rng: RandomNumberGenerator):
+  # 16 random bytes with the bytes on index 6 and 8 modified
+  var b = uuidbinrng(rng)
+
+  return '%02x%02x%02x%02x-%02x%02x-%02x%02x-%02x%02x-%02x%02x%02x%02x%02x%02x' % [
+    # low
+    b[0], b[1], b[2], b[3],
+
+    # mid
+    b[4], b[5],
+
+    # hi
+    b[6], b[7],
+
+    # clock
+    b[8], b[9],
+
+    # clock
+    b[10], b[11], b[12], b[13], b[14], b[15]
+  ]
+  
+var _uuid: Array
+
+class _sentinel extends RandomNumberGenerator:
```

Comment:
Why not use directly `RandomNumberGenerator` ?

Reply:
I created custom class because Godot does not allow using `null` as sentinel together with type hints.

When I was writing the code I was thinking, about avoiding extra object instantiation, but now that I look at it, I failed at that while also adding extra unnecessary complexity. 

Right now I see three options:

1. Create a new RNG object by default (lower performance)
2. Forego type hints and use null (undesirable for me because type hints improve code quality)
3. Use built in RNG by default and create a separate static method that allows passing an instance of RNG

I think 3 would be the best of both worlds, bit maybe a slight performance gain is not worth additional complexity (instancing objects is still very slow).